services/121-service/src/payments/fsp-integration/intersolve-visa/mappers/intersolve-visa-status.mapper.ts

```diff
@@ -0,0 +1,133 @@
+    const csvFilePath = path.join(__dirname, fileName);
+    const fileContent = fs.readFileSync(csvFilePath, { encoding: 'utf-8' });
+    const rows = fileContent.split('\n').filter((row) => row.trim()); // Filters out empty lines
+
+    if (rows.length > 0) {
+      const headerRow = rows.shift();
+
+      // Header row should always be present, this was needed to avoid a type error
+      if (!headerRow) {
+        throw new Error(`Header row not found in CSV file ${fileName}`);
+      }
+      const headers = headerRow
+        .split(';')
+        .map((header) => header.trim().replace(/^"|"$/g, '')); // Remove quotes from headers
+
+      rows.forEach((row) => {
+        const columns = row.split(';').map((column) => column.trim());
+        const rowObject = headers.reduce((obj, nextKey, index) => {
+          obj[nextKey] = columns[index].replace(/^"|"$/g, ''); // Remove quotes from values
+          return obj;
+        }, {});
+
+        const mappingRow: VisaCard121StatusMapInterface = {
+          TokenBlocked: rowObject['TokenBlocked'] === 'TRUE',
+          TokenStatus: rowObject['TokenStatus'],
+          CardStatus: rowObject['CardStatus'],
+          VisaCard121Status: rowObject['VisaCard121Status']
+            ? rowObject['VisaCard121Status']
+            : 'Unknown', // No need to trim, already done
+          VisaCard121StatusExplanation:
+            rowObject['VisaCard121StatusExplanation'],
+          Actions121: rowObject['Actions121'],
+        };
+        this.mapping.push(mappingRow);
+      });
+    }
+
+    this.isMappingLoaded = true;
+  }
+
+  /**
+   * This function determines the status information and actions for a Visa card.
+   * - It first loads the mapping from a CSV file if it hasn't been loaded yet.
+   * - It then finds the row in the mapping that matches the given token blocked status, wallet status, and card status.
+   * - If a matching row is found, it creates an array of actions from the row's Actions121 field, maps each action to a VisaCardAction enum value, and returns an object with the row's status, explanation, and actions.
+   * - If the cards status is missing it will retrun the accosiacted status
+   * - If no matching row is found, it returns an object with an unknown status, a default explanation, and no actions.
+   *
+   * @param {boolean} input.isTokenBlocked - Whether the token is blocked.
+   * @param {IntersolveVisaTokenStatus} input.walletStatus - The status of the wallet.
+   * @param {IntersolveVisaCardStatus | null} input.cardStatus - The status of the card, or null if there is no card.
+   * @returns {VisaCard121StatusInformationAndActions} The status information and actions for the Visa card.
+   */
+  public static determineVisaCard121StatusInformation({
+    isTokenBlocked,
+    walletStatus,
+    cardStatus,
+  }: {
+    isTokenBlocked: boolean;
+    walletStatus: IntersolveVisaTokenStatus; // | null;
+    cardStatus: IntersolveVisaCardStatus | null;
```

Comment:
I do not fully understand when to use which yet.

Reply:
Not sure we have a convention for it, but I prefer undefined whenever possible because it's simply less code. Sometimes that is not possible because we receive `null` from the database. But we can convert a `null` to `undefined` when we need to.

This is a minor thing, which I don't feel strongly about, just felt like calling out that indeed it seems that we have a loose relationship with `null`s and `undefined`s and the way we use them in the codebase (which I think is also a consequence of working without strict checks on for so long).